cmd/gaiko/server.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/taikoxyz/gaiko/internal/flags"
+	"github.com/taikoxyz/gaiko/internal/prover"
+	"github.com/urfave/cli/v2"
+)
+
+type ProveData struct {
+	ProveMode string `json:"prove_mode"` // block, batch, aggregation
+	Input     []byte `json:"input,omitempty"`
+}
+
+type Response struct {
+	Status  string               `json:"status"`
+	Message string               `json:"message"`
+	Proof   prover.ProofResponse `json:"proof"`
+}
+
+type ProveMode int
+
+const (
+	OntakeBlock   ProveMode = iota // 0
+	PacayaBatch                    // 1
+	Aggregation                    // 2
+	Bootstrap                      // 3
+	StatusCheck                    // 4
+	TestHeartBeat                  // 5
+	HeklaBlock                     // 6 deprecated
+)
+
+func proveHandler(ctx context.Context, args *flags.Arguments, sgxProver *prover.SGXProver, w http.ResponseWriter, r *http.Request, proveMode ProveMode) {
+	if r.Method != http.MethodPost {
+		http.Error(w, "POST Only", http.StatusMethodNotAllowed)
+		return
+	}
+
+	// body, err := io.ReadAll(r.Body)
+	// if err != nil {
+	// 	http.Error(w, "Read request failed", http.StatusBadRequest)
+	// 	return
+	// }
+	defer r.Body.Close()
+	contentType := r.Header.Get("Content-Type")
+	if contentType != "application/json" {
+		fmt.Printf("Prove recievied content type: %s\n", contentType)
+		http.Error(w, "Content-Type must be application/json", http.StatusBadRequest)
+		return
+	}
+
+	// call different command according to data.Type
+	var err error
+	var proofResponse prover.ProofResponse
+	switch proveMode {
+	case TestHeartBeat:
+		fmt.Fprintf(args.ProofWriter, "Hello, %s!", "world")
+	case OntakeBlock:
+		err = oneshot(ctx, sgxProver, args)
+		_ = json.Unmarshal(args.ProofWriter.(*bytes.Buffer).Bytes(), &proofResponse)
+	case HeklaBlock:
+		http.Error(w, "Hekla block prove is deprecated", http.StatusBadRequest)
+		err = fmt.Errorf("hekla block prove is deprecated")
+	case PacayaBatch:
+		err = batchOneshot(ctx, sgxProver, args)
+		_ = json.Unmarshal(args.ProofWriter.(*bytes.Buffer).Bytes(), &proofResponse)
+	case Aggregation:
+		err = aggregate(ctx, sgxProver, args)
+		_ = json.Unmarshal(args.ProofWriter.(*bytes.Buffer).Bytes(), &proofResponse)
+	case Bootstrap:
+		err = bootstrap(ctx, sgxProver, args)
+	case StatusCheck:
+		err = check(ctx, sgxProver, args)
+	default:
+		http.Error(w, "Unknown prove mode", http.StatusBadRequest)
+		err = fmt.Errorf("unknown prove mode: %d", proveMode)
```

Comment:
```suggestion
		return
```
can't `http.Error` twice, need return early